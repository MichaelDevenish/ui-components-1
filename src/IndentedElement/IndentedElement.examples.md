# Indented Element

## Basic usage

```jsx
<div>
    <IndentedElement
      depth={0}
    >
      Parent (no depth)
    </IndentedElement>
    <IndentedElement
      depth={1}
    >
      Child A
    </IndentedElement>
    <IndentedElement
      depth={2}
    >
      Child A1
    </IndentedElement>
    <IndentedElement
      depth={1}
    >
      Child B
    </IndentedElement>
    <IndentedElement
      depth={1}
    >
      Child C
    </IndentedElement>
    <IndentedElement
      depth={2}
    >
      Child C1
    </IndentedElement>
    <IndentedElement
      depth={2}
    >
      Child C2
    </IndentedElement>
    <IndentedElement
      depth={3}
    >
      Child C2a
    </IndentedElement>
</div>
```

## Indented select input options

```jsx
initialState = { value: null };

class CustomOption extends React.PureComponent {
  render () {
    const {
      depth,
      label
    } = this.props.option
  
    return <div
      onClick={(event) => {
        event.preventDefault();
        event.stopPropagation();
        this.props.onSelect(this.props.option, event);
      }}
    >
    <IndentedElement depth={depth}>{label}</IndentedElement>
    </div>
  }
}

  <SelectInput
    name='testInput'
    value={state.value}
    options={[
      {
        label: 'Hello World',
        value: 20,
        depth: 0
      },
      {
        label: 'Try selecting me',
        value: 40,
        depth: 1
      }
    ]}
    handleChange={(option) => {
      setState({ value: option.value })
    }}
    optionComponent={CustomOption}
  />
```

## Indented smart list cell content

```jsx
const { ListColumn } = require('../SmartList/ListColumn');

<SmartList
  data={[
    {
      name: 'Blue Card',
      issuingOrganisation: 'Queensland Gov.',
      expiryDate: '11 Nov 2004',
      status: 'Expired',
      depth: 0
    },
    {
      name: 'Green Card',
      issuingOrganisation: 'Western Australia Gov.',
      expiryDate: '05 Feb 2017',
      status: 'Current',
      depth: 1
    }
  ]}
>
  <ListColumn
    size={{
      md: 6,
      lg: 6
    }}
    header='Qualification'
    tooltipText='This is a test tooltip'
    cell={(row) =>
      <IndentedElement
        depth={row.depth}
      >
        <b>{row.name}</b>
        <div>{row.issuingOrganisation}</div>
      </IndentedElement>
    }
  />

  <ListColumn
    size={{
      md: 3,
      lg: 3
    }}
    header='Expiry Date'
    cell={(row) => row.expiryDate}
  />

  <ListColumn
    size={{
      md: 3,
      lg: 3
    }}
    header='Status'
    cell={(row) => row.status}
  />
</SmartList>
```
