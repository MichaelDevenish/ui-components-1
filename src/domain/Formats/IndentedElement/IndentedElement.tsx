import React from 'react'

import {
  IndentedWrapper
} from './style'

export interface IIndentedElement {
  /** Depth level of the element */
  depth: number
}

export class IndentedElement extends React.PureComponent <IIndentedElement> {
  public render (): JSX.Element {
    const {
      depth,
      children
    } = this.props

    return (
      <IndentedWrapper
        depth={depth}
      >
        {children}
      </IndentedWrapper>
    )
  }
}
