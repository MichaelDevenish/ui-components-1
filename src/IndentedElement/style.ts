import React from 'react'
import styled, { StyledComponentClass, css } from 'styled-components'
import { IIndentedElement } from './IndentedElement'

export const IndentedWrapper = styled.div`
  ${(props: IIndentedElement) => props.depth && css`
    padding-left: ${32 * props.depth}px;
  `}
`
