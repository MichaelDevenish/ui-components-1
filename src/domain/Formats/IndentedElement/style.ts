import React from 'react'
import styled, { StyledComponentClass } from 'styled-components'

import { IIndentedElement } from './IndentedElement'

export const IndentedWrapper = styled.div`
  padding-left: ${(props: IIndentedElement) => `${32 * props.depth}px`};
`
