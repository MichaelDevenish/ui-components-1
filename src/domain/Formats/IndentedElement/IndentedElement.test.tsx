import { mount } from 'enzyme'
import React from 'react'

import { IndentedElement } from './IndentedElement'

describe('<IndentedElement />', () => {
  it('should render an element with no indent', () => {
    const wrapper = mount(
      <IndentedElement
        depth={0}
      >
        Hello I am not indented
      </IndentedElement>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should render an indented element', () => {
    const wrapper = mount(
      <IndentedElement
        depth={2}
      >
        WOW! I'm depth level two :)
      </IndentedElement>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
