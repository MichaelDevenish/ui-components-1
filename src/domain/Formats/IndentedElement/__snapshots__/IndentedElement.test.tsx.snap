// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndentedElement /> should render an element with no indent 1`] = `
<IndentedElement
  depth={0}
>
  <styled.div
    depth={0}
  >
    <StyledComponent
      depth={0}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "fvKsMw",
            "rules": Array [
              "
  padding-left: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="fvKsMw"
      >
        Hello I am not indented
      </div>
    </StyledComponent>
  </styled.div>
</IndentedElement>
`;

exports[`<IndentedElement /> should render an indented element 1`] = `
<IndentedElement
  depth={2}
>
  <styled.div
    depth={2}
  >
    <StyledComponent
      depth={2}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "ckkVDv",
            "rules": Array [
              "
  padding-left: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="ckkVDv"
      >
        WOW! I'm depth level two :)
      </div>
    </StyledComponent>
  </styled.div>
</IndentedElement>
`;
